src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -334,7 +336,7 @@ public void Serialize(TextWriter textWriter, object? o)
         [RequiresUnreferencedCode(TrimSerializationWarning)]
         public void Serialize(TextWriter textWriter, object? o, XmlSerializerNamespaces? namespaces)
         {
-            XmlWriter xmlWriter = XmlWriter.Create(textWriter);
+            XmlWriter xmlWriter = XmlWriter.Create(textWriter, s_writerSettings);
```

Comment:
As this comes with a performance hit, should we put this behind an AppContext switch? I would still have this code be the default as correctness takes precedence over performance, but allowing the performance regression to be avoided without any code changes might be useful.